virtool/groups/data.py

```diff
@@ -77,6 +81,11 @@ async def update(
         if not permissions:
             raise ResourceNotFoundError
 
+        user_list = [
```

Comment:
I think this would work better after the database update.

The first reason that comes to mind is that this DB call will be wasted if the update fails for some reason. The only reason you needs to get `users_list` is if you are going to build a response.